### PR TITLE
Notification monitors

### LINKED ISSUE #80: Allow notifications to be bound to any monitor
OSD notifcations can be bound to a specific monitor, or follow the user's active monitor.

Regular notifications should behave the same.

### PR DESCRIPTION
**Pull request by matavach**
Regular notifications now have the same monitor functions as OSD notifcations.

Is there a reason that position options for the two of them differ? Regular Notifications are not allowed on the left / right of the screen, but OSD notifications are.

fixes #80

### CODE REVIEW

File: modules/osd/index.ts [hunk 1]
```diff
@@ -38,20 +38,6 @@ const handleReveal = (self: any, property: string) => {
...
-const getPosition = (pos: OSDAnchor): ("top" | "bottom" | "left" | "right")[] => {
-    const positionMap: { [key: string]: ("top" | "bottom" | "left" | "right")[] } = {
-        "top": ["top"],
-        "top right": ["top", "right"],
-        "top left": ["top", "left"],
-        "bottom": ["bottom"],
-        "bottom right": ["bottom", "right"],
-        "bottom left": ["bottom", "left"],
-        "right": ["right"],
-        "left": ["left"],
```
Comment (Jas-SinghFSU) [inline modules/osd/index.ts:50]: In the process of consolidating the getPosition function, it seems that you forgot to include the `right` and `left` properties.

File: lib/utils.ts [hunk 2]
```diff
@@ -111,3 +113,20 @@ export function createSurfaceFromWidget(widget: Gtk.Widget) {
...
+// currently accepts either notification or OSD, should these be merged?
+*/
+export function getPosition (pos: NotificationAnchor | OSDAnchor): ("top" | "bottom" | "left" | "right")[] {
+    const positionMap: { [key: string]: ("top" | "bottom" | "left" | "right")[] } = {
+        "top": ["top"],
+        "top right": ["top", "right"],
+        "top left": ["top", "left"],
+        "bottom": ["bottom"],
+        "bottom right": ["bottom", "right"],
+        "bottom left": ["bottom", "left"]
```
Comment (Jas-SinghFSU) [inline lib/utils.ts:128]:
```suggestion
        "bottom left": ["bottom", "left"],
        "right": ["right"],
        "left": ["left"],
```

Left and Right anchors seem to be missing. Let's add this here.

File: options.ts [hunk 3]
```diff
@@ -745,6 +745,8 @@ const options = mkOptions(OPTIONS, {
 
     notifications: {
         position: opt<"top" | "top right" | "top left" | "bottom" | "bottom right" | "bottom left">("top right"),
+        monitor: opt(0),
+        active_monitor: opt(true), //unsure if I should make the default true/false. I use false, but OSD's version has default as true.
```
Comment (Jas-SinghFSU) [inline options.ts:749]:
```suggestion
        active_monitor: opt(true),
```
Yes, active monitor as default is ok. User's can disable if desired.

**Comment by matavach** [comment]
I wasn't able to replicate the issue today at all, so I'm guessing it had nothing to do with the code change.
I would like to test it a bit before though.

**Comment by Jas-SinghFSU** [comment]
Sounds good. When you've verified, could you resolve the conversations above? I'll pull and retest afterwards.

File: widget/settings/pages/config/notifications/index.ts [hunk 4]
```diff
@@ -10,6 +10,8 @@ export const NotificationSettings = () => {
         children: [
             Header('Notification Settings'),
             Option({ opt: options.notifications.position, title: 'Notification Location', type: 'enum', enums: ['top left', 'top', 'top right', 'bottom right', 'bottom', 'bottom left'] }),
```
Comment (Jas-SinghFSU) [inline widget/settings/pages/config/notifications/index.ts:12]:
```suggestion
            Option({ opt: options.notifications.position, title: 'Notification Location', type: 'enum', enums: ['top left', 'top', 'top right', 'right', 'bottom right', 'bottom', 'bottom left', 'left'] }),
```

**Comment by Jas-SinghFSU** [comment]
@matavach I've created a branch `notification_position` , if you could merge that into your PR that would be great. I added left and right anchors for notifications and updated the types to reflect that.

And you have one merge conflict that needs to be resolved but that should be fairly quick since it's small.

**Pull request merged**